components/state-component.js

```diff
@@ -0,0 +1,56 @@
+// state-component.js
+
+var _ = require('underscore');
+var jsonpointer = require('jsonpointer');
+var promiseComponent = require('./promise-component');
+
+/**
+ * Stores the most recent packet for every inPort and calls an onchange function
+ * when they change and after when all required port have received something.
+ * If multiple ports both provide an indexBy property name or function,
+ * only packets that both resolve to the same value with be called together
```

Comment:
This provides a hook for vni management. i.e. a function that will extract the vnid from the payload.